code/model/jobs/DNCreateEnvironment.php

```diff
@@ -132,4 +137,24 @@ public function start() {
 		$log->write($message);
 	}
 
+	public function createEnvironment() {
+		$backend = $this->getBackend();
+		if($backend) {
+			return $backend->createEnvironment($this);
+		}
+		throw new Exception("Unable to find backend.");
+	}
+
+	public function getBackend() {
```

Comment:
This points to the fact that the backend functions should live in the Project backend, not in the Environment backend ;-)